src/cms/views/pages/page_view.py

```diff
@@ -245,30 +245,46 @@ def post(self, request, *args, **kwargs):
 
         else:
             # Save forms
-            page_translation_form.instance.page = page_form.save()
-            page_translation_form.save()
-            # Add the success message and redirect to the edit page
-            if not page_instance:
-                messages.success(
-                    request,
-                    _('Page "{}" was successfully created').format(
-                        page_translation_form.instance.title
-                    ),
-                )
-                return redirect(
-                    "edit_page",
-                    **{
-                        "page_id": page_form.instance.id,
-                        "region_slug": region.slug,
-                        "language_slug": language.slug,
-                    },
+            if (page_translation_form.instance.status != status.AUTO_SAVE) or (
+                not (
+                    page_form.instance.id
+                    and (
+                        not Page.objects.filter(id=page_form.instance.id)
+                        .first()
+                        .translations.filter(status=status.PUBLIC)
+                        .exists()
+                    )
                 )
+            ):
+                page_translation_form.instance.page = page_form.save()
+                page_translation_form.save()
+                # Add the success message and redirect to the edit page
+                if not page_instance:
+                    messages.success(
+                        request,
+                        _('Page "{}" was successfully created').format(
+                            page_translation_form.instance.title
+                        ),
+                    )
+                    return redirect(
+                        "edit_page",
+                        **{
+                            "page_id": page_form.instance.id,
+                            "region_slug": region.slug,
+                            "language_slug": language.slug,
+                        },
+                    )
```

Comment:
I don't really understand what's the purpose of this if condition. If I try to translate it into human language, it does the following:

_If the translation status is not `AUTO_SAVE` or if there is no public translation of the page, save both the page translation form and the page form_

But I don't really see how that fixes the issue. This completely ignores the autosaved version if a public version exists (what is not what should happen) and it saves the page form if no public version exists (which should also not happen). Instead, we want that on every autosave, the page translation form is saved (no matter if there exists a public version or not) and the page form should be ignored (with the exception when no page exists at all - then we need the page form to be saved, otherwise the page translation form cannot be saved as well).
```suggestion
            if not page_instance or page_translation_form.instance.status != status.AUTO_SAVE:
                page_translation_form.instance.page = page_form.save()
            # Save page translation form
            page_translation_form.save()
            # Add the success message and redirect to the edit page
            if not page_instance:
                messages.success(
                    request,
                    _('Page "{}" was successfully created').format(
                        page_translation_form.instance.title
                    ),
                )
                return redirect(
                    "edit_page",
                    **{
                        "page_id": page_form.instance.id,
                        "region_slug": region.slug,
                        "language_slug": language.slug,
                    },
                )
```